smarts/core/coordinates.py

```diff
@@ -247,11 +258,13 @@ class Pose:
     """A pair of position and orientation values."""
 
     # TODO: these should be np.ndarray
-    position: Sequence  # [x, y, z]
-    orientation: Sequence  # [a, b, c, d] -> a + bi + cj + dk = 0
+    position: Union[NamedTuple, Tuple, np.ndarray]  # [x, y, z]
+    orientation: Union[
+        NamedTuple, Tuple, np.ndarray
+    ]  # [a, b, c, d] -> a + bi + cj + dk = 0
     heading_: Optional[Heading] = None  # cached heading to avoid recomputing
 
-    def __eq__(self, other) -> bool:
+    def __eq__(self, other: Any) -> bool:
         if not isinstance(other, Pose):
             return False
         return (self.position == other.position).all() and (
```

Comment:
This still seems to be faulty, `all` is not an attribute of `Tuple`, I am surprised we have not encountered errors with this.